api/src/game/game.controller.ts

```diff
@@ -0,0 +1,133 @@
+import {
+  Body,
+  Controller,
+  Get,
+  HttpException,
+  Param,
+  Post,
+  Put,
+} from '@nestjs/common';
+import { AppGateway } from '../app.gateway';
+import { GameState, Player } from '../types';
+import { GameService } from './game.service';
+import { INITIAL_BOARD } from '../constants';
+
+@Controller('game')
+export class GameController {
+  constructor(
+    private readonly gameService: GameService,
+    private gatewayService: AppGateway,
+  ) {}
+
+  @Post('start')
+  async createGame(): Promise<GameState> {
+    const res = await this.gameService.createGame();
+    return this.gameService.deserializer(res);
+  }
+
+  @Get(':id')
+  async getGameState(@Param('id') id: string): Promise<GameState> {
+    try {
+      const res = await this.gameService.getGame({ id: Number(id) });
+      return this.gameService.deserializer(res);
+    } catch (e) {
+      throw new HttpException("That game id doesn't exists", 400);
+    }
+  }
+
+  @Get('join/:id')
```

Comment:
joining a game is an action, you are modifying the state of your game it cannot be a get but a put IMO
and the route shoulbe be /games/:id/join

Reply:
yes, it's done